### ISSUE TITLE
(C4 styled) Sequence diagram

### ISSUE THREAD

**Issue by kirchsth** [plantuml-stdlib/C4-PlantUML#307]
The PlantUML forum contains a [request](https://forum.plantuml.net/17787) that sequence diagrams uses "C4-PlantUML elements".

It is not possible that all C4-PlantUML features are supported in a sequence diagram, but we could try to define participants in the correct styles (other elements like database and boundaries are only very limited supported, e.g. system boundary in following sample is wrong formatted) and based on that the calls,.. could be added "manually".
Additional it have to be checked if the calls can be combined with stereotypes,....

But we could start at least start with a limited version which defines all participants via C4-Stdlib

[![](https://www.plantuml.com/plantuml/svg/pLXTRvj857tdLzojAsthpTh6O20H8GNRx3viLvMSRX_Ig1h3n47X0ysCbQIR_UyxV9YC2m4iMLNE0ubdxhddtBa3OlX06I9i6uKIlGlY514KWOzlq3Pab_Y1sM6mYICSC_3u1HD92BidAFqI-EmMffE9C1wXXu-OKhJ1g-0x1das4Lb_uGsEVJU9sJm94m83DqSjma7UtOOaszWluXX1CUK1t4OzDY4-9lKu8P1WZw5u4sBJ37EEJJ2oBFXN0evr9URLakJpkK1zxZzHWbzPu0KfYfbfqaVAS5G9h1WcE66FAQvN3XpDep-kmK2MDSDP57D2pGDbhhlp2RSZkC649l4hC88ybf1-FeIipQE7--XeqBTrbWoHnygO8ApcQ1GMDu4eBlBgtVZPgkBIjueAYOyYl4uU6cJ5dZSfjtQgVvwNUrvquTZtmiHGfxfJNvU9hZYQsx0DNe6HOcVqyI5cRHuzsCUB_V5uVXJ4c1nxPLJ5M2nc1pqtNuENeaizhOY9cqSFjoBHb0GCmzjYLU1ADHvKOonhrzp8v40SXw2UElEvhKzx4hIbMy9HUX3akIj0xoJ8FBq8eujGv1UWjHDqjmQvZI2xTPos4JIt0Mq4fucWjX2MJGIzXU0smJWcOT54q5e8TdVHtSkwREtqig64ussDvjRDkhRtehbh8g5DOzRd5frL8hRSyo6WwBRZd6houO-PunEcCzqm5A4Smz1K-KM2NbQxEDLKHL7rtNevQYD1NRelmf7h1DrjmsBw8q5psx6KzmcAsmLdd-2udPY91CNjna8adBezC1K8Ypu4JI2uVM0y4-HU1FjGWlFg268DasKVWYuGDBSRIrdSIqRtpb3sjxUipbgX_BIdHd4O_HKkBYuW3E8x2d72W67Ai9-D9t7uABuD0zscQK8OXIku1dWN9u17cr7nJdWo6eqWJIWDrY5-3xzmI28N-L4GW_1-VoMPRyiVqlavsDbyG1b1F9bqdUdp8nlS8bhuGDnY531k3HX-gBoKzcdrzid3Rt9ci8aJWlriMfAoWJUsCr-yAV_dYYiKfI66LNwM-ngRDFYEp-IfTLMCN9lZRCHw8vXkEFcPvlF_qbx5ztZz3QKfxByjNsN5Vi5hiDCq33p4WYIMnkDdzVdW1EpLfz7VoqkOymfWEfBXup_pdP_nM9BiC4pkAMmfzmKi0I-9k3A6Q1ko82lgyZvRAW8iY738bMcseZZsivMHzcsoCdHNM-bnbsAVA1Lj_DsggZk3uV0pnNGu_1gBPVnvUVbf9NWkupcrKioeVs16-G3E_joG9EfA8zq6FXRQBHMVpe04criccGFXq7qY_h2UpFnRsTbWaiCQ3eiFTSEXESvdB7YIdipof6YLqUK1qGBpys_JcUrQLV0kmCrXxVRo2VLGYCz7qva5K0n1MSt4AilO5oo-5rM2nK6dMn2ARdKaBuvJXoUlTcnxVd65An4rHxE8451LiBj_dgNU-OZUOlzzNP8th6Yuz85Vjb7uFm00)](https://www.plantuml.com/plantuml/uml/pLXTRvj857tdLzojAsthpTh6O20H8GNRx3viLvMSRX_Ig1h3n47X0ysCbQIR_UyxV9YC2m4iMLNE0ubdxhddtBa3OlX06I9i6uKIlGlY514KWOzlq3Pab_Y1sM6mYICSC_3u1HD92BidAFqI-EmMffE9C1wXXu-OKhJ1g-0x1das4Lb_uGsEVJU9sJm94m83DqSjma7UtOOaszWluXX1CUK1t4OzDY4-9lKu8P1WZw5u4sBJ37EEJJ2oBFXN0evr9URLakJpkK1zxZzHWbzPu0KfYfbfqaVAS5G9h1WcE66FAQvN3XpDep-kmK2MDSDP57D2pGDbhhlp2RSZkC649l4hC88ybf1-FeIipQE7--XeqBTrbWoHnygO8ApcQ1GMDu4eBlBgtVZPgkBIjueAYOyYl4uU6cJ5dZSfjtQgVvwNUrvquTZtmiHGfxfJNvU9hZYQsx0DNe6HOcVqyI5cRHuzsCUB_V5uVXJ4c1nxPLJ5M2nc1pqtNuENeaizhOY9cqSFjoBHb0GCmzjYLU1ADHvKOonhrzp8v40SXw2UElEvhKzx4hIbMy9HUX3akIj0xoJ8FBq8eujGv1UWjHDqjmQvZI2xTPos4JIt0Mq4fucWjX2MJGIzXU0smJWcOT54q5e8TdVHtSkwREtqig64ussDvjRDkhRtehbh8g5DOzRd5frL8hRSyo6WwBRZd6houO-PunEcCzqm5A4Smz1K-KM2NbQxEDLKHL7rtNevQYD1NRelmf7h1DrjmsBw8q5psx6KzmcAsmLdd-2udPY91CNjna8adBezC1K8Ypu4JI2uVM0y4-HU1FjGWlFg268DasKVWYuGDBSRIrdSIqRtpb3sjxUipbgX_BIdHd4O_HKkBYuW3E8x2d72W67Ai9-D9t7uABuD0zscQK8OXIku1dWN9u17cr7nJdWo6eqWJIWDrY5-3xzmI28N-L4GW_1-VoMPRyiVqlavsDbyG1b1F9bqdUdp8nlS8bhuGDnY531k3HX-gBoKzcdrzid3Rt9ci8aJWlriMfAoWJUsCr-yAV_dYYiKfI66LNwM-ngRDFYEp-IfTLMCN9lZRCHw8vXkEFcPvlF_qbx5ztZz3QKfxByjNsN5Vi5hiDCq33p4WYIMnkDdzVdW1EpLfz7VoqkOymfWEfBXup_pdP_nM9BiC4pkAMmfzmKi0I-9k3A6Q1ko82lgyZvRAW8iY738bMcseZZsivMHzcsoCdHNM-bnbsAVA1Lj_DsggZk3uV0pnNGu_1gBPVnvUVbf9NWkupcrKioeVs16-G3E_joG9EfA8zq6FXRQBHMVpe04criccGFXq7qY_h2UpFnRsTbWaiCQ3eiFTSEXESvdB7YIdipof6YLqUK1qGBpys_JcUrQLV0kmCrXxVRo2VLGYCz7qva5K0n1MSt4AilO5oo-5rM2nK6dMn2ARdKaBuvJXoUlTcnxVd65An4rHxE8451LiBj_dgNU-OZUOlzzNP8th6Yuz85Vjb7uFm00)

 BR Helmut

**Comment by kirchsth** [comment]
nested boxes can be supported with the teoz layout (but stereotype support is still open), arrows supports stereotypes
e.g. 

[![](http://www.plantuml.com/plantuml/svg/jP3DJiCm3CVlVWeppfFea01DMpLMr1VWY3WuRJIgfN7bEk83yUwaOQeYFexuOCT_-nTRterAeg5tS3O87Nf2jVo6Ai42ZCVE3oJKO_JQDTr0NZSRuu8jItm7Z7QlLYphws1hzbgnOy7pEbcg_wxkgJaUX8DlJxMYkBhPtttLM5eh9_sokgwh2_Z8bp7yaYtnJqD8X9__73F9IShIY3zrhSLnlWX0N1KN3VS3U-jr4P6aZC4KKuBvOP5mdlS0s-rjQdYCtlesOXcyMiojFtun2JRVuHXmYa3Bqipq6h4XvmemovBMwRs2NUmDlVi4)](http://www.plantuml.com/plantuml/uml/jP3DJiCm3CVlVWeppfFea01DMpLMr1VWY3WuRJIgfN7bEk83yUwaOQeYFexuOCT_-nTRterAeg5tS3O87Nf2jVo6Ai42ZCVE3oJKO_JQDTr0NZSRuu8jItm7Z7QlLYphws1hzbgnOy7pEbcg_wxkgJaUX8DlJxMYkBhPtttLM5eh9_sokgwh2_Z8bp7yaYtnJqD8X9__73F9IShIY3zrhSLnlWX0N1KN3VS3U-jr4P6aZC4KKuBvOP5mdlS0s-rjQdYCtlesOXcyMiojFtun2JRVuHXmYa3Bqipq6h4XvmemovBMwRs2NUmDlVi4)

**Comment by kirchsth** [comment]
Not all points are solved (no box support, person does not support default sprite, ...) but a possible implementation could look like below

[![](https://www.plantuml.com/plantuml/svg/hLNVSngt4t_dhrZV9FEz-mvmhUkyf2K91XhJsiRZm-aZevEMEurrqbNI6PDE__UkXB5nx9bsEdqJqkvdFpyM-EGyixvhLA_tF-16tw7r83rifA-12I6zD9efUceQfWKegH6OhRe6jOV1z7fsFbxElynMyyl9-SrqDag6-N2OW3VGEGHbE5DgspkIQtXxnwnafS8LtaldNPfysvnaFO0ZgRdg1480odsD3Rgy9OQ3U1vCJbQTbXx5C5mUDhojqA_sG-YWEdnjHau81VxUeUQumpX2vV0GgVQ-TU_ptBBDi28dkfB4MBB7a-eXUP7VIijhvsi26rmffltDnNcEz_GkKFmjw_zorairT2_NlTxZX9_CUYqvvWLQYjVbh6sLv2maktBnxaDlBCI4HZAfqIvPbIORBCTjc_JXZMij4HybBzl2QrbDZ39sz6P--VDYDHrV_qgN2YiYiIIEem04r0fy3vzaOTS4Ecvb6z0WB5PhJOlMItJ0B88s7bpNjiPw592MMo0V0HM6lNFPZ_6ZaAvLR0jEVaNOc4u9A16yCL3AglUe9-NVzo4ffAuK3gvOXJ1-afBGqo_iZXMHJQGorbMdc0qlqranWVddoyNrR1hKUbQvqO4tIkhRKRAFS1TVpAmDEONudHTAbeVH9Hd9NowcY_TmDZk_k5b2QUxtOg6HpX5Je9f70QjJqsd1x3OjYTFuQlvCG0P_z82AczPer3xbnr5jfM6k8uOrIc6KKxGsO5zycS0rEl_DwrWfit4Gri-1D-02XjHqZ6xFjKUhqSCfqxS1fjWwZyqmoPwF_o4CGjvPwRTFzmVZJv55sw-ikPC27QmxpNU_Cw7b6XNpAFOCGeVKVU0rqrNuq3BdDiOArmVqVFYFxAIIFvyMObgc_2IaopmhcSD09awD761_3GMliM4lze3sbwhZLksyxif6UWVSeY2bagbWNroiOd6PdosNLyM_NPR3Bw0rfV77VSUPmj5tm-Dtn2GI8HBHzGbJD5gwSUVhG8G77zCCZDvNKbLKxb0Xz_1_M5lJF2I-gT5YF6lM88pWOvGmFPtilh4yeiS_1jSY9vcGqXtlObmPI2tmFeoPXqDQeDyTaXEMP7rudCqeGKjOg29aahtQKpxlqSOJlHUThzEo3rc2mBNKCXvJWHG-LJf4s1iEBlvoViLzfT6HJR43Uk3oSV21d7Zazew89Sb3zgusraSEEojDvvqK6AWLPulVLkUppxFBQPhrFg4MDF4l)](https://www.plantuml.com/plantuml/uml/hLNVSngt4t_dhrZV9FEz-mvmhUkyf2K91XhJsiRZm-aZevEMEurrqbNI6PDE__UkXB5nx9bsEdqJqkvdFpyM-EGyixvhLA_tF-16tw7r83rifA-12I6zD9efUceQfWKegH6OhRe6jOV1z7fsFbxElynMyyl9-SrqDag6-N2OW3VGEGHbE5DgspkIQtXxnwnafS8LtaldNPfysvnaFO0ZgRdg1480odsD3Rgy9OQ3U1vCJbQTbXx5C5mUDhojqA_sG-YWEdnjHau81VxUeUQumpX2vV0GgVQ-TU_ptBBDi28dkfB4MBB7a-eXUP7VIijhvsi26rmffltDnNcEz_GkKFmjw_zorairT2_NlTxZX9_CUYqvvWLQYjVbh6sLv2maktBnxaDlBCI4HZAfqIvPbIORBCTjc_JXZMij4HybBzl2QrbDZ39sz6P--VDYDHrV_qgN2YiYiIIEem04r0fy3vzaOTS4Ecvb6z0WB5PhJOlMItJ0B88s7bpNjiPw592MMo0V0HM6lNFPZ_6ZaAvLR0jEVaNOc4u9A16yCL3AglUe9-NVzo4ffAuK3gvOXJ1-afBGqo_iZXMHJQGorbMdc0qlqranWVddoyNrR1hKUbQvqO4tIkhRKRAFS1TVpAmDEONudHTAbeVH9Hd9NowcY_TmDZk_k5b2QUxtOg6HpX5Je9f70QjJqsd1x3OjYTFuQlvCG0P_z82AczPer3xbnr5jfM6k8uOrIc6KKxGsO5zycS0rEl_DwrWfit4Gri-1D-02XjHqZ6xFjKUhqSCfqxS1fjWwZyqmoPwF_o4CGjvPwRTFzmVZJv55sw-ikPC27QmxpNU_Cw7b6XNpAFOCGeVKVU0rqrNuq3BdDiOArmVqVFYFxAIIFvyMObgc_2IaopmhcSD09awD761_3GMliM4lze3sbwhZLksyxif6UWVSeY2bagbWNroiOd6PdosNLyM_NPR3Bw0rfV77VSUPmj5tm-Dtn2GI8HBHzGbJD5gwSUVhG8G77zCCZDvNKbLKxb0Xz_1_M5lJF2I-gT5YF6lM88pWOvGmFPtilh4yeiS_1jSY9vcGqXtlObmPI2tmFeoPXqDQeDyTaXEMP7rudCqeGKjOg29aahtQKpxlqSOJlHUThzEo3rc2mBNKCXvJWHG-LJf4s1iEBlvoViLzfT6HJR43Uk3oSV21d7Zazew89Sb3zgusraSEEojDvvqK6AWLPulVLkUppxFBQPhrFg4MDF4l)

It can be tested with my extended branch
BR 
Helmut

**Comment by Potherca** [comment]
Nice!

**Comment by kirchsth** [comment]
new implementation available on my extended branch, now it should support basically all possible options:

### (C4 styled) Sequence diagram

C4-PlantUML **does not offer** a full sequence diagram support,  but existing elements and relationships can be reused as as participants and calls in the corresponding styles

> **!!! Contrary to all other diagrams, please define boundaries without `{` and `}` and mark a boundary end with `Boundary_End()` !!!**

  - Import: `!include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Sequence.puml`
  - Macros (based on **component** diagram macros):
    - Basically all element specific macros (Person, System, Container...) can be reused with following differences:
      - element descriptions are not displayed (?descr argument is ignored); size would be too big
      - element technology/type (?techn/?type) supports no line breaks
      - **boundaries have to be defined without `{` and `}`** and instead of `}` the **`Boundary_End()`** macro has to be called
    - Additional (element specific) Macros:
      - `Boundary_End()`
    - Only following 3 relationship specific macros are supported:
      - `Rel($from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")`
      - `RelIndex($e_index, $from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")`
      - `RelSpecialIndex($rel, $e_index, $from, $to, $label, $techn="", $descr="", $sprite="", $tags="", $link="")`:
        `$rel` enables the definition of all PlantUML specific arrow types, details see e.g. [All arrow types](https://plantuml.com/sequence-diagram#4764f83f72ed032f) and [Slanted or odd arrows](https://plantuml.com/sequence-diagram#5bd6712206960fab)
    - The 5 index related macros (like the dynamic diagram)
      - (lowercase) `increment($offset=1)`: increase current index (procedure which has no direct output)
      - (lowercase) `setIndex($new_index)`: set the new index (procedure which has no direct output)
      - `LastIndex()`: return the last used index (function which can be used as argument)
      - `Index($offset=1)`: returns current index and calculates next index (function which can be used as argument)
      - `SetIndex($new_index)`: returns new set index and calculates next index (function which can be used as argument)

  - (Typically additional used) PlantUML statements:
    - [Grouping message](https://plantuml.com/sequence-diagram#425ba4350c02142c)
    - [Divider or separator](https://plantuml.com/sequence-diagram#d4b2df53a72661cc)
    - [Reference](https://plantuml.com/sequence-diagram#63d5049791d9d79d)
    - [Delay](https://plantuml.com/sequence-diagram#8f497c1a3d15af9e)


[![](http://www.plantuml.com/plantuml/svg/vLVlRziu4lskls9g55ldKDj9cjw7dksiVwYfxx9sO7btKLmE1YMD9I8qgIEfkEcX__iDARl-aJZN5elxiW6QKjIyzsRUa0prix5Cst8fWbVmmkQuH1XSp4l9BQQmq6e9xS55oxymtM0ROSe2TOvQFGuANd2PY395oAqjpBjMIxDLC-CsB-FIe4wKj2XjCr7BrXtNIMvitXfSD6u4a_RNNwvR-8dUfvZIv3p2VvSe4smMBaTASaKySDRfd04J0Wfb38y5Wh4F0WqagXGfnFIC5kuvWtkcEOlTAqfh0NL2dfw08ugLJb6lWO6vux9WcYrXetYbLLdqLVhGPybTHWyo7IYXDBpiJtuT3tlJZ_F-rNmmkPvCZw7VKwM7c6S1Foxr0HbLy_s2teId8KzWfhX0xGO_akWM_HrfRa5LXTyW_0f-6_KWcdsy3cbyUNa97oQ_mw0tXX4Cm_UZSGZJy7qu3SU343w4lM4uXVUJyGmy7AwcOJX-Ff5zgpEDA9y3VAV9GtxFdwrqM-fHyFSul8Oyc-LZJuVXTUyZJEdtB1p2bfLdKcbCWLctGIdu2K4KxE70FmeSJsPmCucYKVzuuohLVc_mZwkfIn6E5UKr3YqKFCjj9RS3AGtsqRB87GHWTOb1q3RkeHkOhnQnp8N-9m187RCV0Hot6U1By2K8zFwYyR05WPAdG0HejzRAJvHqpHSekDnVJhtfTFBx6lu2PSeN40G3Kc0KgUl9sMkeHLnc0XitB4Fe5ONW2RDSoHgz-ZkxPr6YUM61oHHwCYi5q-xDZLPkMHX_pAe5s1nX92rgYHPY9Ycz31QbJ1mL4zm-W5MGbCQg9MhZ3r_2S0qhZ27MQcLGDsidEzdD-wurJ3_KOnBitOnsiwD8W85Q5agYjFNat1UIIKe278DMGg3FD2gqo-MNtmOmHMCFtlQ486dNP6rSXiPnS4d3_PBwwvAY1sDnsJoKV-E4C2ar5RgTtv7l8vDxlkrxCqN1BAQR31o2ozUGv4ncxg5WngoeiUOre4sS_CQSUIZJ-gvjmxYUN93Ia5aMCuDErTDxCTXCGvJak6G-HQiqLXw0nemRgtqqfR1GUkd7ho5d9dSRjoHVf5st0HADgNjaWb9ZIOB6W53PDi-0jFqwgrfKnajkpIuChF9BBPgCMnzcivk8G1v3yTxN0HCKnKrlgunKCYYvYQGeNx91WOc5l_XBnxgdgnmr-h5aIuGEN7grONz0wErTfAqM2vvWAq9zJx0MsowrkV5ptQ2NfbzTdh6iNgEbIqkI-5uQwhB5JkqnpBsDCxyTEzIlEwMffKYJ0ZEKwGm_sOuZ0e92ig7VpysM1KlrcvOx6XfiXKOjiUNBy_UUhZk0KpH-evAotwk5LWLgoz40qmXIMJ1bKIZjbbazVl3B6mKkKLfpyZV_c79JkDFByCy8gyrToIe5CS-26z9JijwZ0j6rPRsNdPt3C1h0w6eycOP3Lx1zACWC9GMNbVUx7ld4gS0rNr_5EspdtqPtyiUSOe-4M6OwEsqbmRjERNDFhUwe_c9ADabBrZUCJGMFT--db9k_Fdw2MljbuMlitifE1mvEi5lPTZw_Epll_hCwLVxLRhcPRWtQRUDdvl5wbtUxxVPs75JnZmx8Mxd3_x_hMQrMpKqXhge__Pdw2hxyWUVgtkhwiPFmHvVWpcxJAbRs_BL9cC3EQVFyxV_pU7q5YMVtNsMcm8JAX3hD9QNlpGbmcU8d9pDoWtg4jXhKBbZjX7GtsWmAfeaBXQSaygSmyJv6AalfxIEFfgNNlOGK5rnoFwodICsdI8C86yEXklPznasaaZwRg29QvtBPw98JNtDxIudLQklUcrnfwtEeh5IbDJn5IkqnJwDRVtDwiajL-sjbpQQ1YLR6UBPRUUkQzjCH4jXdsPfrliS2LBXnpHHuZmBSeaxTt4z4VVX7mFz_U9RJRvfoDdVe1_mACPPfQ_p-9-v2O6N-PiLMNLZm1FpNlI7pJH1uTcXqWHZVGM__vZ5rNJ0s29YmO4f_zLYKhia60m1m88Vr98_GFOelBAO12eDW59LBPn7LkO05TRpK5OTGge2pqzDJi7o99e3rptvcmt44FMjPSkS3K18j_GjOOlazi-YY_6KVr3rggCpp-Rq3ju3sAwN978LGMuJ70ynoxZea8KPGaYOCqaicG729naa28OA0F1yHAxMVV_RLKX--qKplsL7JfcWA9GrwYIaMQFan-lNMOq_BLC7jHgEAPe-Zz-MQpIQyXIMNfKK3xkZGG7E7gdvN-q-B8bUIS15z8SwlmwjmFAIp_6ToYFRlVm40)](http://www.plantuml.com/plantuml/uml/vLVlRziu4lskls9g55ldKDj9cjw7dksiVwYfxx9sO7btKLmE1YMD9I8qgIEfkEcX__iDARl-aJZN5elxiW6QKjIyzsRUa0prix5Cst8fWbVmmkQuH1XSp4l9BQQmq6e9xS55oxymtM0ROSe2TOvQFGuANd2PY395oAqjpBjMIxDLC-CsB-FIe4wKj2XjCr7BrXtNIMvitXfSD6u4a_RNNwvR-8dUfvZIv3p2VvSe4smMBaTASaKySDRfd04J0Wfb38y5Wh4F0WqagXGfnFIC5kuvWtkcEOlTAqfh0NL2dfw08ugLJb6lWO6vux9WcYrXetYbLLdqLVhGPybTHWyo7IYXDBpiJtuT3tlJZ_F-rNmmkPvCZw7VKwM7c6S1Foxr0HbLy_s2teId8KzWfhX0xGO_akWM_HrfRa5LXTyW_0f-6_KWcdsy3cbyUNa97oQ_mw0tXX4Cm_UZSGZJy7qu3SU343w4lM4uXVUJyGmy7AwcOJX-Ff5zgpEDA9y3VAV9GtxFdwrqM-fHyFSul8Oyc-LZJuVXTUyZJEdtB1p2bfLdKcbCWLctGIdu2K4KxE70FmeSJsPmCucYKVzuuohLVc_mZwkfIn6E5UKr3YqKFCjj9RS3AGtsqRB87GHWTOb1q3RkeHkOhnQnp8N-9m187RCV0Hot6U1By2K8zFwYyR05WPAdG0HejzRAJvHqpHSekDnVJhtfTFBx6lu2PSeN40G3Kc0KgUl9sMkeHLnc0XitB4Fe5ONW2RDSoHgz-ZkxPr6YUM61oHHwCYi5q-xDZLPkMHX_pAe5s1nX92rgYHPY9Ycz31QbJ1mL4zm-W5MGbCQg9MhZ3r_2S0qhZ27MQcLGDsidEzdD-wurJ3_KOnBitOnsiwD8W85Q5agYjFNat1UIIKe278DMGg3FD2gqo-MNtmOmHMCFtlQ486dNP6rSXiPnS4d3_PBwwvAY1sDnsJoKV-E4C2ar5RgTtv7l8vDxlkrxCqN1BAQR31o2ozUGv4ncxg5WngoeiUOre4sS_CQSUIZJ-gvjmxYUN93Ia5aMCuDErTDxCTXCGvJak6G-HQiqLXw0nemRgtqqfR1GUkd7ho5d9dSRjoHVf5st0HADgNjaWb9ZIOB6W53PDi-0jFqwgrfKnajkpIuChF9BBPgCMnzcivk8G1v3yTxN0HCKnKrlgunKCYYvYQGeNx91WOc5l_XBnxgdgnmr-h5aIuGEN7grONz0wErTfAqM2vvWAq9zJx0MsowrkV5ptQ2NfbzTdh6iNgEbIqkI-5uQwhB5JkqnpBsDCxyTEzIlEwMffKYJ0ZEKwGm_sOuZ0e92ig7VpysM1KlrcvOx6XfiXKOjiUNBy_UUhZk0KpH-evAotwk5LWLgoz40qmXIMJ1bKIZjbbazVl3B6mKkKLfpyZV_c79JkDFByCy8gyrToIe5CS-26z9JijwZ0j6rPRsNdPt3C1h0w6eycOP3Lx1zACWC9GMNbVUx7ld4gS0rNr_5EspdtqPtyiUSOe-4M6OwEsqbmRjERNDFhUwe_c9ADabBrZUCJGMFT--db9k_Fdw2MljbuMlitifE1mvEi5lPTZw_Epll_hCwLVxLRhcPRWtQRUDdvl5wbtUxxVPs75JnZmx8Mxd3_x_hMQrMpKqXhge__Pdw2hxyWUVgtkhwiPFmHvVWpcxJAbRs_BL9cC3EQVFyxV_pU7q5YMVtNsMcm8JAX3hD9QNlpGbmcU8d9pDoWtg4jXhKBbZjX7GtsWmAfeaBXQSaygSmyJv6AalfxIEFfgNNlOGK5rnoFwodICsdI8C86yEXklPznasaaZwRg29QvtBPw98JNtDxIudLQklUcrnfwtEeh5IbDJn5IkqnJwDRVtDwiajL-sjbpQQ1YLR6UBPRUUkQzjCH4jXdsPfrliS2LBXnpHHuZmBSeaxTt4z4VVX7mFz_U9RJRvfoDdVe1_mACPPfQ_p-9-v2O6N-PiLMNLZm1FpNlI7pJH1uTcXqWHZVGM__vZ5rNJ0s29YmO4f_zLYKhia60m1m88Vr98_GFOelBAO12eDW59LBPn7LkO05TRpK5OTGge2pqzDJi7o99e3rptvcmt44FMjPSkS3K18j_GjOOlazi-YY_6KVr3rggCpp-Rq3ju3sAwN978LGMuJ70ynoxZea8KPGaYOCqaicG729naa28OA0F1yHAxMVV_RLKX--qKplsL7JfcWA9GrwYIaMQFan-lNMOq_BLC7jHgEAPe-Zz-MQpIQyXIMNfKK3xkZGG7E7gdvN-q-B8bUIS15z8SwlmwjmFAIp_6ToYFRlVm40)



@Potherca, @mweagle: I have only one question: Should we defines the sequence diagram styles via variables (like now in all other diagrams) or should we define it only via styles in the themes? What do you think?

**Comment by Potherca** [comment]
Regarding the variables yay/nay... In principle I am in favor of keeping things as consistent as possible but not at _any_ price.
What is the impact?

**Comment by kirchsth** [comment]
do you mean ($offset) instead of (?offset) in the readme or something different?

**Comment by kirchsth** [comment]
**regarding variable**:
a) if we use variables in sequence diagram it is more complex (e.g. If I (re)use the theme `united` as base theme, then I would have to defined all (re)used colors as variables that I can reset it afterwards) and if the theme contains styles (e.g. font names, which are not modeled as variable at all) then I would have to use variables and styles.
In concrete I only updated the missing/"wrong" c4_united styles
```plantuml
' "C4 styled" default is no foot boxes
hide footbox

' fix wrong ReferenceHeaderFontColor
<style>
referenceHeader {
  fontcolor $INFO
}
</style>

' fix invisible SequenceDelayFontColor
<style>
sequenceDiagram {
  delay {
    FontColor $DARK
  }
}
</style>

``` 

b) In the e.g. component diagram I have to use variables, otherwise I could not calculate the legend,... but in the sequence diagram I have no "calculated dependencies"

therefore I end with the simpler solution

**Regarding simple solution:**
In last MR I introduced 3 new relationships Rel(..), RelIndex() and RelSpecialIndex() but I think the simpler solution is that we use only one call with 2 optional parameters Rel(..., ?index, ?special).
In the dynamic diagram I will add the optional ?index too
